dump/match_infra.py

```diff
@@ -0,0 +1,287 @@
+        if len(self.return_fields) > 0:
+            str += "Return Fields: " + ",".join(self.return_fields) + " "
+        if self.ns:
+            str += "Namespace: " + self.ns 
+        if self.match_entire_list:
+            str += "Match Entire List: True "
+        else:
+            str += "Match Entire List: False "
+        return str
+    
+class SourceAdapter:
+    def __init__(self):
+        pass
+    
+    def connect(self, db, ns):
+        return False
+    
+    def getKeys(self, db, table, key_pattern):
+        return []
+    
+    def get(self, db, key):
+        return {}
+    
+    def hget(self, db, key, field):
+        return ""
+    
+    def sep(self, db):
+        return ""
+        
+class RedisSource(SourceAdapter):
+    def __init__(self):
+        self.db_driver = None 
+        
+    def connect(self, db, ns):
+        try:
+            self.db_driver = SonicV2Connector(namespace=ns, host="127.0.0.1")
+            self.db_driver.connect(db)
+        except Exception as e:
+            verbose_print("RedisSource: Connection Failed\n" + str(e))
+            return False
+        return True
+    
+    def sep(self, db):
+        return self.db_driver.get_db_separator(db)
+       
+    def getKeys(self, db, table, key_pattern):       
+        try:
+            keys = self.db_driver.keys(db, table + self.sep(db) + key_pattern)
+        except Exception as e:
+            verbose_print("RedisSource: {}|{}|{} Keys fetch Request Failed for DB {}\n".format(table, self.sep(db), key_pattern, db) + str(e))
+            return []
+        return keys
+    
+    def get(self, db, key):
+        try:
+            fv_pairs = self.db_driver.get_all(db, key)
+        except Exception as e:
+            verbose_print("RedisSource: hgetall {} request failed for DB {}\n".format(key, db) + str(e))
+            return {}
+        return fv_pairs
+     
+    def hget(self, db, key, field):
+        try:
+            value = self.db_driver.get(db, key, field)
+        except Exception as e:
+            verbose_print("RedisSource: hget {} {} request failed for DB {}\n".format(key, field) + str(e))
+            return ""
+        return value
+
+class JsonSource(SourceAdapter):
+    
+    def __init__(self):
+        self.db_driver = None
+    
+    def connect(self, db, ns):
+        try:
+            with open(db) as f:
+                self.db_driver = json.load(f)
+        except Exception as e:
+            verbose_print("JsonSource: Loading the JSON file failed" + str(e))
+            return False
+        return True
+    
+    def sep(self, db):
+        return SonicDBConfig.getSeparator("CONFIG_DB")
+    
+    def getKeys(self, db, table, key_pattern):
+        if table not in self.db_driver:
+            return []
+        
+        all_keys = self.db_driver[table].keys()
+        key_ptrn = key_pattern
+        key_ptrn = re.escape(key_ptrn)
+        key_ptrn = key_ptrn.replace("\\*", ".*")
```

Comment:
Does this work for any redis keys pattern expression?

Reply:
Made changes to handle all the glob patterns, as specified here [Link](https://redis.io/commands/KEYS)